WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java

```diff
@@ -136,6 +136,17 @@ private void updateUIAndNote(boolean doRefresh) {
             return;
         }
 
+        // here compare the current Note - is it any different from the Note the user is
+        // currently viewing?
+        // if it is, then replace the dataset with the new one.
+        // If not, just let it be.
+        if (mAdapter != null) {
+            Note currentNote = mAdapter.getNoteWithId(mNoteId);
```

Comment:
Is there a reason `mAdapter.getNoteWithId(mNoteId)` was used rather than `NotificationsTable.getNoteById(mNoteId)` to get the current note?  If `NotificationsTable.getNoteById(mNoteId)` is used instead, you could combine it with the previous conditional to make something like this:
```
        Note note = NotificationsTable.getNoteById(mNoteId);
        if (note == null) {
            // no note found
            showErrorToastAndFinish();
            return;
        } else if (note.similar(currentNote)) {
            return;
        }
```